src/qt/buynamespage.cpp

```diff
@@ -72,13 +119,91 @@ void BuyNamesPage::onNameEdited(const QString &name)
     }
 }
 
-void BuyNamesPage::onRegisterNameAction()
+void BuyNamesPage::onAsciiNameEdited(const QString &name)
 {
     if (!walletModel)
         return;
 
-    QString name = ui->registerName->text();
+    try
+    {
+        const QString hexName = ASCIIToHex(name);
+        const QString domainName = ASCIIToDomain(name);
+
+        ui->registerNameHex->setText(hexName);
+        ui->registerNameDomain->setText(domainName);
+
+        availableError();
+    }
+    catch(InvalidNameString &e)
+    {
+        ui->statusLabel->setText(tr("Not a valid ASCII entry!"));
+    }
+
+}
+
+void BuyNamesPage::onHexNameEdited(const QString &name)
+{
+
+    if (!walletModel)
+        return;
+
+
+    try{
+        const QString asciiName = HexToASCII(name);
+        const QString domainName = ASCIIToDomain(asciiName);
+        
+        ui->registerNameAscii->setText(asciiName);
+        ui->registerNameDomain->setText(domainName);
+    
+        availableError();
+    }
+    catch(InvalidNameString &e)
+    {
+        ui->statusLabel->setText(tr("Not a valid entry!"));
+    }
+
+}
+
+void BuyNamesPage::onDomainNameEdited(const QString &name){
+    
+    if (!walletModel)
+        return;
+
+    try
+    {
+        const QString asciiName = DomainToASCII(name);
+        const QString hexName = ASCIIToHex(asciiName);
+
+        ui->registerNameAscii->setText(asciiName);
+        ui->registerNameHex->setText(hexName);
+        
+        if(IsPurportedNamecoinDomain(name.toStdString()))
+        {
+            availableError();
+        }
+        else
+        {
+            ui->statusLabel->setText(tr("%1 is not a valid Namecoin domain!").arg(name));
```

Comment:
Might be worth putting `%1` inside double-quotes so that it's a bit less ambiguous. Also can we make the text a bit more helpful? Is this just checking for the `.bit` suffix? If so, maybe say in the error text that the `.bit` suffix is expected?